Radix-related fixes, improved main menu + OSM sign-in

opyh opened the pull request:
- Fix change requests from https://github.com/sozialhelden/wheelmap-frontend/pull/459
- Fix <kbd>L</kbd> hotkey (this key unlocks the OSM sign-in menu item)
- Refactor main menu and sign-in with OSM to use Radix components

<img width="529" alt="Screenshot of the new Radix UI-based main menu" src="https://github.com/user-attachments/assets/0cdfb114-8cba-4a01-85b5-d95b45e7e70b" />

kriskbx commented on src/components/App/MainMenu/SessionMenuItem.tsx:
```diff
@@ -0,0 +1,69 @@
+import { signIn, signOut, useSession } from "next-auth/react";
```
I really really love how this section turned out. It looks amazing. Great work 🫶 

![A man, probably from South-America, sitting in front of the camera, saying "Mucho mucho joy"](https://media1.tenor.com/m/Qe4HyPPaoFAAAAAd/ted-lasso-tedlassogifs.gif)

kriskbx reviewed (changes requested):
Great work, thanks a lot for your amazing PR ❤️ I really like the direction, the interface is going now. 😻 

**Bug:** One thing that needs to be fixed though: after opening the dropdown menu either through the mouse or by focusing it and using enter/space I cannot navigate the items with the arrow keys. After pressing an arrow key, the error "TypeError: Cannot read properties of null (reading 'focus')" gets thrown. Can you take a look at this?

**Discussion/Suggestion:** I like that all those navigation items got moved into a dropdown menu, it makes the interface more clear and removes distractions. However, this also means that users will have a harder time finding things. This can have some negative implications especially when we want them to find some things very easily. For example when we want them to participate, e.g. the button to "add a new place". I suggest to move the "add a new place" button out of the dropdown menu again right next to menu button and also make sure it visually pops out more, e.g. by using the color "gray" for the menu button and the primary color for the "add a new place" button. Maybe we could also use an icon instead the text "add a new place". I don't know, but worth trying out. I'm curious on your opinion @opyh and @Mayaryin, what do you think? Does that make sense? Do you have other ideas?

opyh commented on biome.json:
```diff
@@ -14,7 +14,7 @@
 	"formatter": {
 		"enabled": true,
-		"indentStyle": "tab"
+		"indentStyle": "space"
```
Makes sense! I could create a new PR for this behavior.

kriskbx replied:
I actually wanted to express my feeling of joy by posting that gif. I like the change. But yeah, let's do another PR. Fine for me 😄

opyh commented on src/components/App/MainMenu/MainMenu.tsx:
```diff
@@ -1,315 +1,107 @@
-      </StyledNav>
-    </Card>
-	);
+        <VectorImage
```
Absolutely. We need logos in 2 variants for this to work. Will create an extra PR for this.

opyh commented on src/components/App/MainMenu/MainMenu.tsx:
```diff
@@ -1,315 +1,107 @@
+            <Flex justify={"between"} align="center" p="2">
+              <Flex align="center" gap="4">
+                {logoHomeLink}
+                <Text as="div">{claimString}</Text>
```
You're right, this shouldn't even be visible on mobiles!

opyh commented on src/components/App/MainMenu/link-types/JoinedEventLink.tsx:
```diff
@@ -0,0 +1,37 @@
+  const label = joinedMappingEvent ? joinedMappingEvent.name : props.label;
+  const children = isValidating ? <Spinner /> : <Text>{label}</Text>;
+  const router = useRouter();
+  // XXX: This should be an <AppStateAwareLink>, but this would break keyboard navigation:
```
This is the fix for the keyboard-related JS crash you mentioned.

opyh commented on src/components/App/MapLayout.tsx:
```diff
@@ -50,34 +52,36 @@ export default function MapLayout({
+  const { toggleExpertMode } = useExpertMode()
+  const expertModeHotkeys: HotkeyConfig[] = React.useMemo(() => [
+    {
+      combo: 'mod+e',
```
Switched this to <kbd>Cmd</kbd>+<kbd>E</kbd> to avoid accidental activation.

opyh commented on src/components/App/MapLayout.tsx:
```diff
@@ -50,34 +52,36 @@ export default function MapLayout({
-    setIsMenuOpen(typeof newValue === 'boolean' ? newValue : !isMenuOpen)
-  }, [isMenuOpen])
+  const router = useRouter()
+  const isOnboardingVisible = firstStart || router.pathname === '/onboarding'
```
This is a part of the fix to get the missing onboarding dialog back when clicking the logo in the upper left.

opyh commented on src/components/CombinedFeaturePanel/PlaceLayout.tsx:
```diff
@@ -30,11 +30,11 @@ export default function PlaceLayout({
   return (
     <Toolbar>
-      <Link href={closeUrl ?? '/'}>
+      <AppStateLink href={closeUrl ?? '/'}>
```
Keeps the map state when closing the place details.

opyh commented on src/lib/fetchers/ac/refactor-this/fetchMappingEvent.ts:
```diff
@@ -1,10 +1,10 @@
 export function useMappingEvent(_id: string) {
-  const response = useDocumentSWR<MappingEvent>({
-    collectionName: 'MappingEvents',
+  const response = useDocumentSWR({
```
The API here wasn't correctly used before.

opyh commented on src/lib/util/useViewportSize.ts:
```diff
@@ -1,34 +1,32 @@
  * Determine the viewport size depending on the inner sizing of the window, useful
  * for handling layout choices depending on the size of the browser
  */
 export const useWindowSize = () => {
```
The changes here fix hydration errors and make server-side only renderings use 1024x768 as fallback window size.

opyh commented on src/components/App/MainMenu/link-types/SessionElement.tsx:
```diff
@@ -0,0 +1,64 @@
+
+function NonAuthenticatedMenuContent() {
+  const handleSignIn = React.useCallback(() => signIn("osm"), []);
+  return <Button onClick={handleSignIn}>{t`Sign in`}</Button>;
```
Shortened link as simple fix for the too-long-text issue.

opyh commented on src/components/App/MainMenu/MainMenuLinks.tsx:
```diff
@@ -0,0 +1,63 @@
+        <DropdownMenu.Trigger>
+          <Button
+            variant="solid"
+            color="gray"
```
New styling 💅

opyh commented:
Implemented the improvements you wished for (and hopefully some more!)

kriskbx commented on src/components/App/MainMenu/link-types/JoinedEventLink.tsx:
```diff
@@ -0,0 +1,37 @@
+  const children = isValidating ? <Spinner /> : <Text>{label}</Text>;
+  const router = useRouter();
+  // XXX: This should be an <AppStateAwareLink>, but this would break keyboard navigation:
+  // A bug in Radix UI (?) appears to not update the internal `ref` correctly.
```
**Non-Blocker:** Maybe the AppStateLink needs to use forwardRef in order to be able to be used with ref? Idk, my knowledge of modern react is still a bit limited, but that came to mind. Nothing that we need to solve in this PR.

opyh replied:
That was it! I was 100% sure I had tried this, but missed it.

kriskbx commented on src/components/Map/filterLayers.ts:
```diff
@@ -164,7 +164,7 @@ export function filterLayers(
       layout: omit(localizedLayer.layout, 'line-z-offset'),
       paint: omit(localizedLayer.paint, 'line-occlusion-opacity'),
     }
-    console.log(enhancedLayer)
```
Thank you 😄

kriskbx reviewed (changes requested):
Hey, thanks again for your amazing work all those nice changes ❤️  Unfortunately I still found some issues. 😭 

The "Events" Link is not a link but a span-tag. It also takes a few seconds after clicking before something happens without any immediate feedback. This is not accessible and causes other issues down the line. I would consider this a blocker, that needs to be changed before merging this PR.
![image](https://github.com/user-attachments/assets/e3dd7c7c-f72e-4bd4-9c5b-179fa5d924b2)

**Non-Blocker:** The dropdown menu doesn't behave as expected. When opening the menu and pressing the down arrow, I would expect that the first item is highlighted. Unfortunately, that doesn't work. I can press the up-arrow and then the last item in the list is highlighted and I can navigate the list. This is a weird behaviour. It's still somewhat accessible and should not be blocking this PR, but something we should fix in the near future.
![keyboard-nav-issue](https://github.com/user-attachments/assets/7f6bbb2e-311b-486b-9089-9ab74e9f8a63)

opyh commented:
Thanks for your thorough testing! I've moved the sign-in button out of the menu, added the missing `forwardRef()` wrapper, and arrow keys appear to work as expected now.

Could you have another look?

kriskbx reviewed (approved):
Thanks again for your amazing work. LGTM 🫶
